metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -59,11 +59,11 @@ class KubernetesDecorator(StepDecorator):
         Number of CPUs required for this step. If `@resources` is
         also present, the maximum value from all decorators is used.
     memory : int, default 4096
-        Memory size (in MB) required for this step. If
+        Memory size (in MiB) required for this step. If
         `@resources` is also present, the maximum value from all decorators is
         used.
     disk : int, default 10240
-        Disk size (in MB) required for this step. If
+        Disk size (in MiB) required for this step. If
```

Comment:
I had a peek at the units, and it seems that memory/disk values for kubernetes are treated as MB so the earlier docstring for these was correct.

we could hold off on changing the kubernetes units in the docs for now, as this would require also changing the implementation to match.

Reply:
alternatively changing the units from `M` to `Mi` in `kube_utils.py#qos_requests_and_limits` should be the only additional change required, as this is used both by the kubernetes and argo workflows implementations.

Reply:
> alternatively changing the units from `M` to `Mi` in `kube_utils.py#qos_requests_and_limits` should be the only additional change required, as this is used both by the kubernetes and argo workflows implementations.

Cool, thanks for taking a look at the Kubernetes units. Since the defaults are multiples of 1024, MiB seems to be the original intent. So I think it makes sense to update the implementation, especially if doing so is as easy as you've described. Accordingly, I've pushed up a new commit: bfaa672a. But, if you prefer, I could revert bfaa672a and 4fc51071 and update the `@resources` docstring to indicate that the memory units are MiB for AWS Batch and MB for Kubernetes. Just let me know your preference. Thanks!

Reply:
Fwiw, the Kubernetes "Assign Memory Resources to Containers and Pods" [example](https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource/) uses MiB, potentially indicating that MiB is more standard / expected there. Maybe.